From 8ab1c3b019e465bb0551ebdf679fd38793c136f5 Mon Sep 17 00:00:00 2001
From: Jarno Jakonen <jarno.jakonen@intel.com>
Date: Mon, 20 Apr 2015 14:51:46 +0300
Subject: [PATCH 07/10] gc2235: Fix corruption in 1296x736 resolution

Adjust cropping window dimensions to avoid garbage at the edges in
online video mode.

Change-Id: I28b59250ca12c3818f0c1f6d0f99246c95a95bff
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-31103
Signed-off-by: Jarno Jakonen <jarno.jakonen@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/53006
Build: IMIN (SC)
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Shuguang Gong <shuguang.gong@intel.com>
Reviewed-by: Shuguang Gong <shuguang.gong@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/gc2235.h | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
index 71a4e72..3ec9ff6 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
@@ -304,7 +304,7 @@ static struct gc2235_reg const gc2235_1296_736_30fps[] = {
 	{ GC2235_8BIT, 0x8c, 0x02 },
 
 	{ GC2235_8BIT, 0x07, 0x01 }, /* VBI */
-	{ GC2235_8BIT, 0x08, 0x48 },
+	{ GC2235_8BIT, 0x08, 0x44 },
 	{ GC2235_8BIT, 0x09, 0x00 }, /* row start */
 	{ GC2235_8BIT, 0x0a, 0xf0 },
 	{ GC2235_8BIT, 0x0b, 0x00 }, /* col start */
@@ -312,15 +312,15 @@ static struct gc2235_reg const gc2235_1296_736_30fps[] = {
 	{ GC2235_8BIT, 0x0d, 0x02 }, /* win height 736 */
 	{ GC2235_8BIT, 0x0e, 0xf0 },
 	{ GC2235_8BIT, 0x0f, 0x05 }, /* win width: 1296 */
-	{ GC2235_8BIT, 0x10, 0x10 },
+	{ GC2235_8BIT, 0x10, 0x20 },
 
 	{ GC2235_8BIT, 0x90, 0x01 },
-	{ GC2235_8BIT, 0x92, 0x02 },
-	{ GC2235_8BIT, 0x94, 0x06 },
+	{ GC2235_8BIT, 0x92, 0x08 },
+	{ GC2235_8BIT, 0x94, 0x08 },
 	{ GC2235_8BIT, 0x95, 0x02 }, /* crop win height 736 */
-	{ GC2235_8BIT, 0x96, 0xd0 },
+	{ GC2235_8BIT, 0x96, 0xe0 },
 	{ GC2235_8BIT, 0x97, 0x05 }, /* crop win width 1296 */
-	{ GC2235_8BIT, 0x98, 0x00 },
+	{ GC2235_8BIT, 0x98, 0x10 },
 	/* mimi init */
 	{ GC2235_8BIT, 0xfe, 0x03 }, /* switch to P3 */
 	{ GC2235_8BIT, 0x01, 0x07 },
@@ -329,7 +329,7 @@ static struct gc2235_reg const gc2235_1296_736_30fps[] = {
 	{ GC2235_8BIT, 0x06, 0x80 },
 	{ GC2235_8BIT, 0x11, 0x2b },
 	/* set mipi buffer */
-	{ GC2235_8BIT, 0x12, 0x40 }, /* val_low = (width * 10 / 8) & 0xFF */
+	{ GC2235_8BIT, 0x12, 0x54 }, /* val_low = (width * 10 / 8) & 0xFF */
 	{ GC2235_8BIT, 0x13, 0x06 }, /* val_high = (width * 10 / 8) >> 8 */
 
 	{ GC2235_8BIT, 0x15, 0x12 }, /* DPHY mode*/
-- 
1.9.1

